contracts/BridgeProxy.sol

```diff
@@ -11,9 +11,16 @@ contract BridgeProxy is Pausable {
     MvxTransaction[] private pendingTransactions;
     mapping(uint256 => TokenPayment) private payments;
     uint256 private lowestTxId;
+    uint256 private currentTxId;
 
     constructor(address _multiTransferAddress) Pausable() {
         multiTransferAddress = _multiTransferAddress;
         lowestTxId = 1;
     }
+
+    function deposit(MvxTransaction calldata txn) external payable whenNotPaused {
+        require(msg.sender == multiTransferAddress, "BridgeProxy: Only MultiTransfer can do deposits");
+        pendingTransactions.push(txn);
+        payments[currentTxId++] = TokenPayment(txn.token, txn.amount);
```

Comment:
for solidity it does not make sense to store also the payments, only the txn is enough